Sources/SpeziBluetooth/CoreBluetooth/BluetoothPeripheral.swift

```diff
@@ -218,22 +221,30 @@ public actor BluetoothPeripheral: BluetoothActor { // swiftlint:disable:this typ
         peripheral.delegate = delegate
     }
 
-    /// Establish a connection to the peripheral.
+    /// Establish a connection to the peripheral and wait until it is connected.
     ///
     /// Make a connection to the peripheral.
     ///
-    /// - Note: This method returns as soon as the request to connect was processed locally. It does
-    ///     not wait till the connection was completed successfully.
-    ///
     /// - Note: You might want to verify via the ``AdvertisementData/isConnectable`` property that the device is connectable.
-    public func connect() {
-        guard let manager else {
-            logger.warning("Tried to connect an orphaned bluetooth peripheral!")
-            return
-        }
-
-        manager.assumeIsolated { manager in
-            manager.connect(peripheral: self)
+    public func connect() async throws {
+        try await connectAccess.waitCheckingCancellation()
+
+        try await withTaskCancellationHandler {
+            try await withCheckedThrowingContinuation { continuation in
+                assert(connectContinuation == nil, "connectContinuation was unexpectedly not nil")
+                connectContinuation = continuation
+                guard let manager = self.manager else {
+                    logger.warning("Tried to connect an orphaned bluetooth peripheral!")
+                    return
+                }
```

Comment:
I would propose to make this check at the very beginning of the connect method. It's an easy and fast check and if it fails we avoid creating all the continuations for nothing and avoid potential programming mistakes.

In its current form it also leaks the continuation (as it never resumes it) and also forgets to reset the `connectContinuation` property. If you just move it to the top you don't need to deal with any of these edge cases 👍